core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiServiceV2.scala

```diff
@@ -31,7 +30,17 @@ trait BillingApiServiceV2 extends UserInfoDirectives {
   val userServiceConstructor: UserInfo => UserService
   val spendReportingConstructor: UserInfo => SpendReportingService
 
-  implicit def spendReportingAggregationKeyUnmarshaller: Unmarshaller[String, SpendReportingAggregationKey] = Unmarshaller.strict(SpendReportingAggregationKeys.withName)
+  implicit def aggregationKeyParameterUnmarshaller: Unmarshaller[String, SpendReportingAggregationKeyWithSub] = Unmarshaller.strict { parameter =>
+    val delimitedParameter = parameter.split("~").toList
+
+    delimitedParameter match {
+      case key :: subKey :: Nil => SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.withName(key), Option(SpendReportingAggregationKeys.withName(subKey)))
```

Comment:
If we are in the first case, we know `subKey` exists. So why is it wrapped with Option?

Reply:
`SpendReportingAggregationKeyWithSub` expects `subKey` to have the type `Option[SpendReportingAggregationKey]` because we may or may not have a sub-aggregation. So in this case, even though we know it exists, we still need to wrap it in `Option` to satisfy the class signature of `SpendReportingAggregationKeyWithSub`. 

We could use `Some` instead of `Option` here, but I like to default to `Option` over `Some` because `Option` is resilient to an unexpected `null` (even though a `null` is theoretically impossible here)

Reply:
I see. I probably would have used `Some`, but I don't feel strongly about it.